rust-src/concordium_base/src/ffi_helpers/common.rs

```diff
@@ -1,8 +1,20 @@
 #[no_mangle]
 /// Free an array that was converted to a pointer from a vector.
 /// This assumes the vector's capacity and length were the same.
+#[deprecated(
+    note = "use [`free_array_len_cap`] instead since it correctly frees the whole capacity size \
+            of the vector."
+)]
```

Comment:
Don't put the deprecation notice on this. It has no effect since this function is not to be used from Rust. None of the `ffi` module is meant to be used from Rust.

Reply:
Also, I think the `free_array_len` is technically correct the way we use it. Ultimately to free the pointer it only needs the pointer, the memory allocator will know what chunk of data has been allocated with that pointer.

So we don't want to make it look like the pervasive use of free_array_len in the node is incorrect.

But free_array_len_cap is simpler to understand so it's cleaner to use for the idiss case.

Reply:
removed